wgpu-core/src/instance.rs

```diff
@@ -134,20 +135,11 @@ impl<B: hal::Backend> Adapter<B> {
             adapter_features.contains(hal::Features::SAMPLER_ANISOTROPY),
         );
         extensions.set(
-            wgt::Extensions::SAMPLED_TEXTURE_BINDING_ARRAY,
-            adapter_features.contains(hal::Features::TEXTURE_DESCRIPTOR_ARRAY),
-        );
-        extensions.set(
-            wgt::Extensions::SAMPLED_TEXTURE_ARRAY_DYNAMIC_INDEXING,
-            adapter_features.contains(hal::Features::SHADER_SAMPLED_IMAGE_ARRAY_DYNAMIC_INDEXING),
-        );
-        extensions.set(
-            wgt::Extensions::SAMPLED_TEXTURE_ARRAY_NON_UNIFORM_INDEXING,
-            adapter_features.contains(hal::Features::SAMPLED_TEXTURE_DESCRIPTOR_INDEXING),
-        );
-        extensions.set(
-            wgt::Extensions::UNSIZED_BINDING_ARRAY,
-            adapter_features.contains(hal::Features::UNSIZED_DESCRIPTOR_ARRAY),
+            wgt::Extensions::BINDING_INDEXING,
+            adapter_features.contains(
```

Comment:
This shouldn't require `UNSIZED_DESCRIPTOR_ARRAY`, right? Either we need to only check for the basic hal feature here, or change `contains` to `intersects` ;)

Reply:
**mutters under breath** yup it's intersects...